.github/workflows/ghcr.yml

```diff
@@ -34,11 +52,63 @@ jobs:
           tags: |
             type=sha
             type=ref,event=tag
-            ${{ github.event_name == 'push' && 'type=raw,value=main' || '' }}
+            type=raw,value=main,enabled=${{ github.event_name == 'push' }}
+            type=raw,value=main
+
+      - name: Get relevant tag for internal image version
+        id: get-version
+        shell: python
+        run: |
+          import json
+          metadata = json.loads('${{ steps.metadata.outputs.json }}')
+          version = metadata["labels"]["org.opencontainers.image.version"]
+          if version == "main":
+              # get first version that isn't main
+              for tag in metadata["tags"]:
+                  if ":main" not in tag:
+                      version = tag.split(":")[1]
+
+          print(f"::set-output name=version::{version}")
 
       - uses: docker/build-push-action@v3
         with:
           context: .
           tags: ${{ steps.metadata.outputs.tags }}
           labels: ${{ steps.metadata.outputs.labels }}
           push: true
+          build-args: |
+            RELEASE=${{ steps.get-version.outputs.version }}
+
+  deploy-staging:
+    name: Deploy staging frontend
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
```

Comment:
Will this deploy to staging on each push in a pull request (to main)?

Reply:
No because the push event is scoped to the main branch.

You can confirm this is true by seeing that this workflow step was skipped in this PR: https://github.com/WordPress/openverse-frontend/runs/7283335174?check_suite_focus=true

Reply:
I see, makes sense! Though then is the `on: -> pull_request: ...` scope doing something in this workflow? 🤔
I remember we used to publish an image for each PR but seems that has been lost at some point.